### DISCUSSION TITLE
Changing cache owner count in embedded cluster

### DISCUSSION THREAD

**Discussion by anand-kulk** [infinispan/infinispan#12550] [Q&A] [answered]
Hi- I have embedded ispn cluster using  14.0.1.Final in kubernetes with 5 pods and 2 cache owners for distributed caches. I changed configuration to up owner count to 3 by doing a rolling restart and hence preserving cache entries. To test if there are 3 replicas for all cache entries, I abruptly deleted 2 pods of 5. My assumption is cluster should withstand loss of `no of owners -1` nodes. But, to my surprise I lost portion of cache entries. 
So, I have questions - 

1. Does changing owners on live cluster, only applies to new cache entries or it should re-balance existing entries to match new owner count?
2. Do I need to take the cluster offline for this change to take effect? Clearly, applying config with rolling restart didn't work.
3. How about other cache timeout settings like remote-timeout, lock-acquistion-timeout, can they take effect without downtime?

Thanks.

**Comment by tristantarrant** [comment]
Hi,
unfortunately ownership is not a mutable attribute, so a restart is necessary.
The mutable attributes for cache configurations are:
```
aliases
clustering.remote-timeout
expiration.interval
expiration.lifespan
expiration.max-idle
indexing.indexed-entities
jmx-statistics.statistics
locking.acquire-timeout
memory.max-count
memory.max-size
persistence.connection-attempts
security.authorization.roles
tracing.categories
tracing.enabled
transaction.single-phase-auto-commit
transaction.stop-timeout
```

**Reply by anand-kulk** [reply]
@tristantarrant  I did rolling restart of one node at a time to apply owner change, isn't that good enough?

**Reply by pruivo** [reply, marked as answer]
As Tristan mentioned, the number of owners is not mutable. Your configuration change wouldn't have any effect and you would need to delete the cache (losing data) and re-created it again.

**Reply by anand-kulk** [reply]
Thanks @pruivo and @tristantarrant
